### PR TITLE
[API] as logged in user, I can fetch detail keyword search result

### LINKED ISSUE #16: [API] as logged in  user, I can fetch detail keyword search result
## Why

Users should be able to fetch detail keyword search result

- The API route should be `GET api/v1/keywords/:id`
- Users can't able to fetch other user's keyword details
- Users can't able to fetch other keyword details if the keyword searching has not been completed
 
## Who Benefits?

Users

### PR DESCRIPTION
**Pull request by junan**
https://github.com/junan/elixir_google_scraper/projects/1#card-61317070

## What happened

Users can fetch keyword search result detail
 
## Insight

N/A
 
## Proof Of Work

Screen Shot 2021-07-08 at 2 04 03 AM
Screen Shot 2021-07-08 at 2 04 16 AM

### CODE REVIEW

File: lib/elixir_google_scraper_web/controllers/api/v1/keyword_controller.ex [hunk 1]
```diff
@@ -41,6 +42,18 @@ defmodule ElixirGoogleScraperWeb.Api.V1.KeywordController do
     end
   end
 
+  def show(conn, %{"id" => id}) do
```
Comment (hanam1ni) [inline lib/elixir_google_scraper_web/controllers/api/v1/keyword_controller.ex:34]: Let's check if keyword is belong to the requested user

File: test/elixir_google_scraper_web/controllers/api/v1/keyword_controller_test.exs [hunk 2]
```diff
@@ -118,4 +118,45 @@ defmodule ElixirGoogleScraperWeb.Api.V1.KeywordControllerTest do
              }
     end
   end
+
+  describe "GET show/2" do
+    test "returrns 200 status with details keyword search result", %{conn: conn} do
```
Comment (hanam1ni) [inline test/elixir_google_scraper_web/controllers/api/v1/keyword_controller_test.exs]:
```suggestion
    test "returns 200 status with details keyword search result", %{conn: conn} do
```

File: test/elixir_google_scraper_web/controllers/api/v1/keyword_controller_test.exs [hunk 3]
```diff
@@ -118,4 +118,45 @@ defmodule ElixirGoogleScraperWeb.Api.V1.KeywordControllerTest do
...
+                 },
+                 "id" => _,
+                 "relationships" => %{},
+                 "type" => "keyword_search_result"
+               },
+               "included" => []
+             } = json_response(conn, 200)
+    end
+
+    test "returrns 404 status with an error message", %{conn: conn} do
```
Comment (hanam1ni) [inline test/elixir_google_scraper_web/controllers/api/v1/keyword_controller_test.exs]:
```suggestion
    test "returns 404 status with an error message", %{conn: conn} do
```

File: lib/elixir_google_scraper_web/controllers/api/v1/keyword_controller.ex (hunk 1 above)
Comment (junan) [reply lib/elixir_google_scraper_web/controllers/api/v1/keyword_controller.ex:34]: Done, API will respond 404 error if the keyword does not exist in the database or keyword does not belong to the user

File: lib/elixir_google_scraper_web/controllers/api/v1/keyword_controller.ex [hunk 4]
```diff
@@ -3,6 +3,8 @@ defmodule ElixirGoogleScraperWeb.Api.V1.KeywordController do
 
   alias ElixirGoogleScraper.Scraper.Keywords
   alias ElixirGoogleScraperWeb.V1.{ErrorView, KeywordView}
+  alias ElixirGoogleScraperWeb.V1.KeywordSearchResultView
+  alias ElixirGoogleScraperWeb.V1.KeywordView
```
Comment (rosle) [inline lib/elixir_google_scraper_web/controllers/api/v1/keyword_controller.ex]:
```suggestion
  alias ElixirGoogleScraperWeb.V1.{ErrorView, KeywordSearchResultView, KeywordView}
```

File: test/elixir_google_scraper/scraper/keywords_test.exs [hunk 5]
```diff
@@ -87,4 +87,36 @@ defmodule ElixirGoogleScraper.Scraper.KeywordsTest do
...
+      user2 = insert(:user)
+      keyword = insert(:keyword, user: user)
+      insert(:search_result, keyword: keyword)
+
+      result = Keywords.get_user_keyword(user2, keyword.id)
+
+      assert result == nil
+    end
+
+    test "returns nil when the given keyword does not exist" do
```
Comment (hanam1ni) [inline test/elixir_google_scraper/scraper/keywords_test.exs:114]: Let's fix all the warnings and error in this test case

File: test/elixir_google_scraper_web/controllers/api/v1/keyword_controller_test.exs [hunk 6]
```diff
@@ -123,4 +123,68 @@ defmodule ElixirGoogleScraperWeb.Api.V1.KeywordControllerTest do
              }
     end
   end
+
+  describe "GET show/2" do
+    test "returns 200 status with details keyword search result when the given keyword does belong to the current user",
```
Comment (rosle) [inline test/elixir_google_scraper_web/controllers/api/v1/keyword_controller_test.exs]:
```suggestion
    test "returns 200 status with details keyword search result when the given keyword belongs to the current user",
```

File: test/elixir_google_scraper/scraper/keywords_test.exs [hunk 7]
```diff
@@ -87,4 +87,36 @@ defmodule ElixirGoogleScraper.Scraper.KeywordsTest do
       end
     end
   end
+
+  describe "get_user_keyword/2" do
+    test "returns the keyword with preloded search result when the given keyword does belong to the user" do
```
Comment (rosle) [inline test/elixir_google_scraper/scraper/keywords_test.exs]:
```suggestion
    test "returns the keyword with preloded search result when the given keyword belongs to the user" do
```

**Review by rosle — APPROVED** [review]

**Review by hanam1ni — APPROVED** [review]

**Pull request merged**